### ISSUE TITLE
TPM2 Quote takes an Algorithm where it should take a Scheme

### ISSUE THREAD

**Issue by chrisfenner** [google/go-tpm#201]
https://github.com/google/go-tpm/blob/ac5b427c06a867a3f94077b406507ec93f1e04e1/tpm2/tpm2.go#L936

The last parameter, `sigAlg`, is an `Algorithm` which corresponds to `TPM_ALG` in the spec, but it should be some type that corresponds to `TPMT_SIG_SCHEME` in the spec. As written, the only parameter that can be passed to `Quote` successfully is `AlgNull` (meaning you have to set the scheme on the AK when you create it).

This is what the test does:

https://github.com/google/go-tpm/blob/ac5b427c06a867a3f94077b406507ec93f1e04e1/tpm2/test/tpm2_test.go#L1258-L1279

**Comment by josephlr** [comment]
We could try to create a scheme based on the Alg, but that seems like a difficult proposition (why key length do we use for RSASSA for instance).

The real solution would be having the params for all functions match the Spec directly. Maybe we should open a tracking issue.

**Comment by chrisfenner** [comment]
Yeah, I don't see how to fix this without a frustrating breaking change to the API.

Long term, I would love to reflect the spec into the API. In the short term, it may be good to fix this and eat the breaking change, or simply document that it only supports AlgNull and that you have to create an AK with the signing scheme that you want.
